src/js/components/modals/JobFormModal.js

```diff
@@ -150,29 +155,92 @@ class JobFormModal extends mixin(StoreMixin) {
   }
 
   handleJSONEditorChange(jsonDefinition) {
-    try {
-      let job = new Job(JSON.parse(jsonDefinition));
-      this.setState({errorMessage: null, job});
-    } catch (error) {
-      // TODO: DCOS-7734 Handle error
-    }
+
+    // Keep the new JSON string in the JSON definition
+    // Validation will occur upon submission of the form
+    this.setState({
+      errorMessage: null,
+      jsonDefinition
+    });
+
   }
 
   handleSubmit() {
-    let {isEdit, job} = this.props;
-    let jobSpec = JobUtil.createJobSpecFromJob(this.state.job);
+    let {isEdit} = this.props;
```

Comment:
Mainly it was just copy-past black magic. However I decided to have a thorough look and properly implement the entire logic. Check the updates.